data_model/src/isi.rs

```diff
@@ -4,865 +4,953 @@
+        pub struct Log {
+            /// Message log level
+            #[serde(flatten)]
+            pub level: Level,
+            #[getset(skip)] // TODO: Fix this by addressing ffi issues
+            /// Msg to be logged
+            pub msg: String,
         }
     }
 }
 
-isi! {
-    /// Sized structure for all possible on-chain configuration parameters.
-    #[derive(Display)]
-    #[display(fmt = "SET `{parameter}`")]
-    #[serde(transparent)]
-    #[repr(transparent)]
-    // SAFETY: `SetParameterExpr` has no trap representation in `EvaluatesTo<Parameter>`
-    #[ffi_type(unsafe {robust})]
-    pub struct SetParameterExpr {
-        /// The configuration parameter being changed.
-        #[serde(flatten)]
-        pub parameter: EvaluatesTo<Parameter>,
-    }
+macro_rules! isi_box {
+    ($($meta:meta)* $item:item) => {
+        #[derive(
+            Debug,
+            Clone,
+            PartialEq,
+            Eq,
+            PartialOrd,
+            Ord,
+            Display,
+            parity_scale_codec::Decode,
+            parity_scale_codec::Encode,
+            serde::Deserialize,
+            serde::Serialize,
+            iroha_schema::IntoSchema,
+            derive_more::From,
+        )]
+        $($meta)*
+        $item
+    };
 }
 
-isi! {
-    /// Sized structure for all possible on-chain configuration parameters when they are first created.
-    #[derive(Display)]
-    #[display(fmt = "SET `{parameter}`")]
-    #[serde(transparent)]
-    #[repr(transparent)]
-    // SAFETY: `NewParameterExpr` has no trap representation in `EvaluatesTo<Parameter>`
-    #[ffi_type(unsafe {robust})]
-    pub struct NewParameterExpr {
-        /// The configuration parameter being created.
-        #[serde(flatten)]
-        pub parameter: EvaluatesTo<Parameter>,
+isi_box! {
+    /// Enum with all supported [`SetKeyValue`] instructions.
+    pub enum SetKeyValueBox {
+        /// Set key value for [`Domain`].
+        Domain(SetKeyValue<Domain>),
+        /// Set key value for [`Account`].
+        Account(SetKeyValue<Account>),
+        /// Set key value for [`AssetDefinition`].
+        AssetDefinition(SetKeyValue<AssetDefinition>),
+        /// Set key value for [`Asset`].
+        Asset(SetKeyValue<Asset>),
     }
 }
 
-isi! {
-    /// Sized structure for all possible key value set instructions.
-    #[derive(Display)]
-    #[display(fmt = "SET `{key}` = `{value}` IN `{object_id}`")]
-    #[ffi_type]
-    pub struct SetKeyValueExpr {
-        /// Where to set this key value.
-        #[serde(flatten)]
-        pub object_id: EvaluatesTo<IdBox>,
-        /// Key string.
-        pub key: EvaluatesTo<Name>,
-        /// Object to set as a value.
-        pub value: EvaluatesTo<Value>,
+impl SetKeyValueBox {
+    /// Constructs a new [`SetKeyValueBox`] for a [`Domain`] with the given `key` and `value`.
+    pub fn domain(domain_id: <Domain as Identifiable>::Id, key: Name, value: Value) -> Self {
```

Comment:
```suggestion
    pub fn domain(domain_id: <Domain as Identifiable>::Id, key: Name, value: impl Into<Value>) -> Self {
```

do this for all `Value`s. This makes user's life a lot easier

Reply:
> But also will bloat size of code.

yes, but that's a price to be paid for nicer API. For any API that is not exposed to users I would say don't use `impl Into<T>`, but for API exposed to users prefer using generics

Reply:
It's a double-edged sword. Having a lot of generic may confuse the user (e. g. some of our `client` methods).

But in this case I agree, will do